Mathlib/Algebra/Order/Floor/Semiring.lean

```diff
@@ -504,3 +504,71 @@ theorem subsingleton_floorSemiring {R} [Semiring R] [LinearOrder R] :
   cases H₁
   cases H₂
   congr
+
+namespace Mathlib.Meta.NormNum
+
+open Qq
+
+/-!
+### `norm_num` extesion for `Nat.floor`
+-/
+
+theorem IsNat.natFloor {R : Type*} [Semiring R] [LinearOrder R] [IsStrictOrderedRing R]
+    [FloorSemiring R] (r : R) (m : ℕ) : IsNat r m → IsNat (⌊r⌋₊) m := by
+  rintro ⟨⟨⟩⟩
+  exact ⟨by simp⟩
+
+theorem IsInt.natFloor {R : Type*} [Ring R] [LinearOrder R] [IsStrictOrderedRing R]
+    [FloorSemiring R] (r : R) (m : ℕ) : IsInt r (.negOfNat m) → IsNat (⌊r⌋₊) 0 := by
+  rintro ⟨⟨⟩⟩
+  exact ⟨Nat.floor_of_nonpos <| by simp⟩
+
+theorem IsNNRat.natFloor {R : Type*} [Semifield R] [LinearOrder R] [IsStrictOrderedRing R]
+    [FloorSemiring R] (r : R) (n d : ℕ) (h : IsNNRat r n d) : IsNat ⌊r⌋₊ (n / d) := by
```

Comment:
Comparing to [isNat_add](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Tactic/NormNum/Basic.html#Mathlib.Meta.NormNum.isNat_add), it looks like `n / d` is supposed to be abstracted to ensure that it can be a raw nat literal.

Reply:
@digama0, does this sound right to you?

Reply:
Yes. (PS: you dropped some indentation)